src/App/App.tsx

```diff
@@ -39,56 +41,75 @@ const App: React.SFC = () => {
 	]);
   const [error, setError] = useState("");
 
+	useEffect(() => {getCocktail()}, []);
+	useEffect(() => {fetchAllCocktails()}, []);
+	useEffect(() => {
+		updateAllCocktails()
+	}, [allCocktails]);
+
   // API Calls
   const fetchAllCocktails = async (): Promise<any> => {
     try {
       const data: AllCocktailsDetails[] = await getAllCocktails();
-      return setAllCocktails(data);
+			return setAllCocktails(data);
     } catch (error) {
-      setError(error.message);
+      setAllCError(error.toString());
     }
   };
 
-	const getCocktail = async ():Promise<any> => {
+	const getCocktail = async ():Promise<void> => {
 		try {
 			const data: Cocktail = await getRandomCocktail();
-			setRandomCocktail(data);
+			return setRandomCocktail(data);
 		} catch (error) {
-			setError(error.message);
+			setRandomCError(error.toString());
 		}
 	};
 
-	useEffect(() => {getCocktail()}, []);
-	useEffect(() => {fetchAllCocktails()});
+	const updateAllCocktails = async ():Promise<void> => {
+		try {
+			const newCocktails = await Promise.all(
+				allCocktails.map(c => getCocktailDetails(c.idDrink))
+			);
+			setAllCocktails(newCocktails);
+		} catch (error) {
+			setError(error.message);
+		}
+	}
 
 	// Functions
 	const findResults = (searchValue: string) => {
-		let searchResults: any = [{
-			strDrink: "",
-			strDrinkThumb: "",
-			idDrink: "",
-		}];
-		allCocktails.forEach(cocktail => {
-			if (cocktail.strDrink.toLowerCase().includes(searchValue.toLowerCase())) {
-				searchResults.push(cocktail);
-			} 
+		const byName = searchByName(searchValue);
+		const byIngredient = searchByIngred(searchValue);
+		const searchedResults = byName.concat(byIngredient);
+		setFilteredResults([...searchedResults]);
+	}
+
+	const searchByName = (keyword: string) => {
+		return allCocktails.filter(cocktail => {
+			return cocktail.strDrink.toLowerCase().includes(keyword.toLowerCase())
+		});
+	}
+
+	const searchByIngred = (keyword: string) => {
+		return allCocktails.filter((cocktail: Cocktail) => {
+			const values = Object.values(cocktail);
+			let result = values.find((value: string | null) => {
+					if (value) return value.toLowerCase() === keyword.toLowerCase();
+			})
+			if (result) return cocktail;
 		});
-		setFilteredResults(searchResults.splice(1));
+		// setFilteredResults(searchResults.splice(1));
 	}
```

Comment:
search by name or ingredient block (line 81-102)

Reply:
Yay! Great job on this!